### DISCUSSION TITLE
Error: read ECONNRESET at TCP.onStreamRead (node:internal/stream_base_commons:217:20)

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1701] [Q&A]
I am getting the below error on vendure server. Please help to fix this error.

Vendure Version: 1.6.4
Deployed in DigitalOcean

ecom-core] [2022-08-03 10:47:40] error 8/3/22, 10:47 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:47:43] error 8/3/22, 10:47 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:47:46] error 8/3/22, 10:47 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:47:53] error 8/3/22, 10:47 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:47:53] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:47:53]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:47:54] error 8/3/22, 10:47 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:47:54] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:47:54]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:47:54] error 8/3/22, 10:47 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:47:54] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:47:54]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:11] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:11] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:12] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "update-search-index" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:15] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "update-search-index" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:18] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:18] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:24] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:28] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "update-search-index" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:29] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:30] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:32] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "update-search-index" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:32] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:33] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:34] error 8/3/22, 10:48 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:48:34] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:48:34]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:34] error 8/3/22, 10:48 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:48:34] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:48:34]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:35] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "update-search-index" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:38] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "apply-collection-filters" encountered an error (set log level to Debug for trace): read ECONNRESET 
[ecom-core] [2022-08-03 10:48:38] error 8/3/22, 10:48 AM - [Vendure Server] Failed to update SelfRefreshingCache "ChannelService.allChannels": read ECONNRESET
[ecom-core] [2022-08-03 10:48:38] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:48:38]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:38] error 8/3/22, 10:48 AM - [Vendure Server] Failed to update SelfRefreshingCache "ChannelService.allChannels": read ECONNRESET
[ecom-core] [2022-08-03 10:48:38] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:48:38]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:38] error 8/3/22, 10:48 AM - [Vendure Server] read ECONNRESET
[ecom-core] [2022-08-03 10:48:38] Error: read ECONNRESET
[ecom-core] [2022-08-03 10:48:38]     at TCP.onStreamRead (node:internal/stream_base_commons:217:20) 
[ecom-core] [2022-08-03 10:48:41] error 8/3/22, 10:48 AM - [Vendure Worker] Job queue "send-email" encountered an error (set log level to Debug for trace): Connection terminated unexpectedly

**Comment by michaelbromley** [comment]
Root cause error seems to be `ECONNRESET` - could it be that a job is taking a very long time?
